routers/user/oauth_userinfo.go

```diff
@@ -0,0 +1,29 @@
+package user
+
+import (
+	"fmt"
+
+	"code.gitea.io/gitea/modules/context"
+)
+
+// UserInfoResponse represents a successful userinfo response
+type UserInfoResponse struct {
+	Sub      string `json:"sub"`
+	Name     string `json:"name"`
+	Username string `json:"preffered_username"`
+	Email    string `json:"email"`
+	Picture  string `json:"picture"`
+}
+
+// UserInfoOAauth responds with OAuth formatted userinfo
+func UserInfoOAuth(ctx *context.Context) {
+	user := ctx.User
+	response := &UserInfoResponse{
+		Sub:      fmt.Sprint(user.ID),
+		Name:     user.FullName,
+		Username: user.Name,
+		Email:    user.Email,
+		Picture:  user.Avatar,
+	}
+	ctx.JSON(200, response)
```

Comment:
a tiny nit, could you use `http.StatusOK` instead of hardcoding 200 here? You'll need to import the `net/http` package as well.

Reply:
well I did this for the API once in the past ... but not for the normal webUI funcs ...

I think it should be done but you could start with it now ... for me this is no blocking thing but a nice to have :)

Reply:
There has been a refactor PR cleaning this up elsewhere, so I think you can use net/http now